collector/socket/src/main/java/org/apache/karaf/decanter/collector/socket/SocketCollector.java

```diff
@@ -60,20 +60,21 @@
     private Dictionary<String, Object> properties;
     private String eventAdminTopic;
     private EventAdmin dispatcher;
+    
+    @Reference
     private Unmarshaller unmarshaller;
```

Comment:
`unmarshaller` should be public (or at least protected or default to used in tests).